metricbeat/module/nats/util/util.go

```diff
@@ -0,0 +1,108 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package util
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+)
+
+// convertUptimeToSeconds converts uptime from formatted string to seconds
+// input: "1y20d22h3m30s", output: 33343410
+func convertUptimeToSeconds(uptime string) (seconds int64, err error) {
```

Comment:
I was wondering why not using `ParseDuration`, then I found this interesting discussion: https://github.com/golang/go/issues/11473#issuecomment-117430007, where it is decided than anything beyond hours is not accurate for durations.
Then I wondered why we needed to parse durations with days and years coming from a go application, and then I found that they are not using `time.Duration` and they have their own formatter that assumes that the duration of all days is 24 hours, and the duration of all years 365 days https://github.com/nats-io/nats-server/blob/958c04840ce2a0b5e435dd7eff27eb0a0f3a0043/server/monitor.go#L1098
:man_shrugging:

The good thing is that parsing these uptimes with the same assumptions gives the original, accurate seconds.

Maybe we could add a comment with a link to the code generating these uptimes.

We could also consider parsing this with a regexp, but not to be done in this PR.